src/deps_deploy/deps_deploy.clj

```diff
@@ -82,6 +88,71 @@
 (defn- artifact [{:keys [group-id artifact-id version]}]
   (str group-id "/" artifact-id "-" version))
 
+(defn- read-edn-files
+  "Given as options map, use tools.deps.alpha to read and merge the
+  applicable `deps.edn` files, from depstar"
+  [{:keys [repro] :or {repro true}}]
+  (let [{:keys [root-edn user-edn project-edn]} (t/find-edn-maps)]
+    (t/merge-edns (if repro
+                    [root-edn project-edn]
+                    [root-edn user-edn project-edn]))))
+
+(defn- set-settings-builder
+  "Copied from clojure.tools.deps.alpha"
+  [^DefaultMavenSettingsBuilder default-builder settings-builder]
+  (doto (.. default-builder getClass (getDeclaredField "settingsBuilder"))
+    (.setAccessible true)
+    (.set default-builder settings-builder)))
+
+(defn- get-settings
+  "Copied from clojure.tools.deps.alpha"
+  ^Settings []
+  (.buildSettings
+    (doto (DefaultMavenSettingsBuilder.)
+      (set-settings-builder (.newInstance (DefaultSettingsBuilderFactory.))))))
+
+(defn- get-repo-settings
+  [repo mvn-repos]
+  (when (contains? mvn-repos repo)
+    (let [repo-settings (->> (get-settings)
+                             (.getServers)
+                             (filter #(= repo (.getId ^Server %)))
+                             first)]
+      (assoc-some (get mvn-repos repo)
```

Comment:
Maybe we could inline `assoc-some` as well, so we don't have to pull in medley as a dep?